src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -2360,6 +2360,9 @@ internal enum ErrorCode
         ERR_ImplicitlyTypedParamsParameter = 9272,
         ERR_VariableDeclarationNamedField = 9273,
         ERR_CollectionExpressionKeyValuePairNotSupported = 9274,
+        ERR_CollectionArgumentsMustBeFirst = 9275,
```

Comment:
nit: You may want to leave a gap between errorcodes in `main` branch and those in dictionary expression branch, so that you don't have to re-number too much (hopefully only once)